src/main.rs

```diff
@@ -376,6 +376,30 @@ async fn enter_redraw_loop(
 					None => needs_redraw = false,
 					Some(action) => match action {
 						InputAction::Redraw => (),
+						InputAction::Click { col, row } => {
+							if let Some((page_num, device_x, device_y)) = tui.map_click_to_page(col, row, &main_area, font_size) {
+								let (resp_tx, resp_rx) = flume::bounded(1);
+								to_renderer.send(RenderNotif::QueryLinkAt {
+									page: page_num,
+									device_x_px: device_x,
+									device_y_px: device_y,
+									resp: resp_tx
+								})?;
+								// await the renderer reply
+								match resp_rx.recv_async().await {
+									Ok(Some(LinkTarget::Uri(uri))) => {
+										if let Err(e) = webbrowser::open(&uri) {
+											log::error!("Failed to open uri {}: {:?}", uri, e);
+										}
+									}
+									Ok(Some(LinkTarget::GoTo { page_index })) => {
+										to_renderer.send(RenderNotif::JumpToPage(page_index))?;
+										to_converter.send(ConverterMsg::GoToPage(page_index))?;
+									}
+									_ => (),
```

Comment:
This error also shouldn't be swallowed; I doubt the channel will fail, but just in case it does, we want to either show that to the user or panic (if it requires that)